File: frontweb/src/types/post.ts
export interface PagedResponse {
  pageInfo: PageInfo;
  data: Post[];
}

export interface Response {
  data: Post[];
}

export interface PageInfo {
  page: number;
  size: number;
  totalPages: number;
  totalElements: number;
  isLast: boolean;
  isFirst: boolean;
  hasNext: boolean;
  hasPrevious: boolean;
}

export interface Post {
  postId: string;
  title: string;
  description: string;
  url: string;
  source: string;
  pubDate: string;
  readTime: string;
  tags: string[];
  thumbnail?: string;
  bookmarked: boolean;
  bookmarkCount: number;
  liked: boolean;
  likeCount: number;
  comments: number;
}

File: blogpostproject/blogpost-domain/src/main/kotlin/com/hobeen/domain/post/Post.kt
package com.hobeen.domain.post

import com.hobeen.domain.tag.Tag
import java.time.LocalDateTime

class Post (
    val postId: PostId,
    val title: String,
    val source: String,
    val url: String,
    val pubDate: LocalDateTime,
    val description: String,
    val thumbnail: String,
    val tags: List<Tag>
)